### ISSUE TITLE
Fio-plot only allows specific metrics to be plotted - users can't select the metrics they want to plot

### ISSUE THREAD

**Issue by louwrentius** [louwrentius/fio-plot#69]
Fio-plot can deal with:

- Bandwidth
- IOPs
- Latency
- Histogram

If you want to plot any other data found inside the JSON file, that's currently not possible.
Why is this:

- Fio-plot is maybe my second Python application ever written and also by far the largest one
- Only the above data/fields are statically mapped from the JSON file into a flat dictionary for further graphing
- The graphs are all hard-coded and are only aware of (some of) the above metrics.

The desired state would be that the user selects the metrics they want from the JSON to be graphed.
There are a few challenges:

1. This does mean that the application should be able to understand the unit type of the metric.
2. Some metrics consists of lists or internal dictionaries that needs to be understood/parsed.

Please note that I'm not a professional developer by any means and this would be a significant undertaking. It would probably mean very significant rework of a lot of components. It must be understood that there is therefore no timeline for this improvement.

**Comment by louwrentius** [comment]
This issue is related to issue #67 where this limitation is mentioned.

**Comment by louwrentius** [comment]
**feedback or ideas on how to deal with this are much appreciated**

**Comment by jsrolon** [comment]
I believe there should be a significant investment in changing the way graphs are specified. IMO the current CLI-based way to interact with fio-plot is cumbersome as it is, and it would become even more unwieldy if users were to be able to specify their own graphs. Should #34 be reconsidered for both bench_fio and fio_plot?

**Comment by louwrentius** [comment]
@jsrolon yes I can see that. I would not remove the CLI option (personal preference) but a configuration file makes sense. In that case I think the same format as FIO uses (ini) is good enough, but other ideas are welcome.

**Comment by louwrentius** [comment]
@jsrolon In the latest release I've also added support for INI configuration for fio-plot so #34 is now addressed.
